controllers/ocirepository_controller.go

```diff
@@ -433,7 +434,41 @@ func (r *OCIRepositoryReconciler) reconcileSource(ctx context.Context, obj *sour
 			return sreconcile.ResultEmpty, e
 		}
 
-		blob, err := layers[0].Compressed()
+		var layer gcrv1.Layer
+
+		switch {
+		case obj.Spec.LayerSelector != nil && obj.Spec.LayerSelector.MediaType != "":
```

Comment:
Might be better to move this to a method on the API object to ensure the `nil` check is always in place.